### PR TITLE
DOC: Partial fix SA04 errors in docstrings #28792 (feedback needed)

### PR DESCRIPTION
**Pull request by dilex42**
- [x] xref #28792

Would like feedback on a few issues before continuing to work on this issue :
- File pandas/core/accessor.py . How to handle multiline text? Couldn't find anywhere good recomendations.
- File pandas/core/ops/docstrings.py . Changed logic of templates. Is this okay? Also new lines with formatted text are too long, would like a reccomended sollution for this.
- File pandas/core/window/rolling.py and some others in regard to window functions have broken links that can be fixed by adding pandas. prefix to them(as in changed code). Is this right thing to do and should I make this changes whenever i see this?

Sorry for obvious mistakes, just starting to contribute. Thanks.

### CODE REVIEW

File: pandas/core/accessor.py [hunk 1]
```diff
@@ -282,7 +282,8 @@ def decorator(accessor):
 @doc(
     _register_accessor,
     klass="DataFrame",
-    others="register_series_accessor, register_index_accessor",
+    others=("register_series_accessor : Register a custom accessor on Series objects.\n"
+            "register_index_accessor : Register a custom accessor on Index objects."),
```
Comment (datapythonista) [inline pandas/core/accessor.py]: Instead of doing this, I prefer to simply repeat the three methods in all docstrings. In each docstring we'll have a self-reference in the See Also, but that's not a big deal. Much better that all this complexity here.

File: pandas/core/ops/docstrings.py [hunk 2]
```diff
@@ -33,9 +33,14 @@ def _make_flex_doc(op_name, typ):
...
             equiv=equiv,
-            reverse=op_desc["reverse"],
             series_returns=op_desc["series_returns"],
         )
+        if op_desc["reverse"]:
+            doc_see_also = _see_also_reverse_SERIES.format(
+                reverse=op_desc["reverse"],
+                see_also_desc=op_desc["see_also_desc"],
+            )
+            doc_no_examples = doc_no_examples + doc_see_also
```
Comment (datapythonista) [inline pandas/core/ops/docstrings.py]: Do you mind opening a separate pull request for this? I think the rest can be merged already, and this is a bit trickier.

**Review by datapythonista — COMMENTED** [review]
Thanks for the work on this @dilex42 

I think with the two suggested changes we can get most of this merged. We can discuss any remaining question you have in the new PR.

**Comment by dilex42** [comment]
@datapythonista Thanks for the feedback. Will do the changes.
Before merging this changes there is still a problem with docs to window functions. The references doesn't work without pandas prefix but validation script gives errors if prefix present. I guess there is some inconsistency either in doc generator or validator.
Also a general rule how to deal with multiline text would be appreciated.

**Comment by datapythonista** [comment]
Can you open an issue for the prefix on the validation. Leave them the way they render correctly. If the validation is breaking the CI, remove the validation of that specific error in `ci/code_checks.sh` please.

For the multiline, I think the best is to use whatever makes code more readable. I think in most cases would be a multiline string, but a `\n` can be better for simple things. But it's more a mix of common sense and personal preference than a rule. Always keeping readability in mind.

**Comment by dilex42** [comment]
Thanks

File: pandas/core/accessor.py [hunk 3]
```diff
@@ -280,9 +282,7 @@ def decorator(accessor):
 
 
 @doc(
-    _register_accessor,
-    klass="DataFrame",
-    others="register_series_accessor, register_index_accessor",
+    _register_accessor, klass="DataFrame",
```
Comment (datapythonista) [inline pandas/core/accessor.py]: Is black formatting this in three lines, even if you write it in one? I think with one would be clearer.

File: pandas/core/arrays/categorical.py [hunk 4]
```diff
@@ -416,12 +416,12 @@ def categories(self):
...
-        add_categories
-        remove_categories
-        remove_unused_categories
-        set_categories
+        rename_categories : Rename categories.
+        reorder_categories : Reorder categories.
+        add_categories : Add new categories.
+        remove_categories : Remove the specified categories.
+        remove_unused_categories : Remove categories which are not used.
+        set_categories : Set the categories to the specified categories.
```
Comment (datapythonista) [inline pandas/core/arrays/categorical.py]: Not sure if this reads a bit better not repeating the word categories?
```suggestion
        set_categories : Set the categories to the specified ones.
```

File: pandas/plotting/_misc.py [hunk 5]
```diff
@@ -29,7 +29,7 @@ def table(ax, data, rowLabels=None, colLabels=None, **kwargs):
 
 def register():
     """
-    Register Pandas Formatters and Converters with matplotlib.
+    Register pandas Formatters and Converters with matplotlib.
```
Comment (datapythonista) [inline pandas/plotting/_misc.py]: I'd use `formatters and converters` in lowercase everywhere, since you're fixing the capitalization too.
```suggestion
    Register pandas formatters and converters with matplotlib.
```

File: pandas/plotting/_misc.py [hunk 6]
```diff
@@ -29,7 +29,7 @@ def table(ax, data, rowLabels=None, colLabels=None, **kwargs):
...
 
     This function modifies the global ``matplotlib.units.registry``
     dictionary. Pandas adds custom converters for
```
Comment (datapythonista) [inline pandas/plotting/_misc.py]:
```suggestion
    dictionary. pandas adds custom converters for
```

File: pandas/plotting/_misc.py [hunk 7]
```diff
@@ -62,7 +62,8 @@ def deregister():
 
     See Also
     --------
-    register_matplotlib_converters
+    register_matplotlib_converters : Register pandas Formatters and Converters
```
Comment (datapythonista) [inline pandas/plotting/_misc.py]:
```suggestion
    register_matplotlib_converters : Register pandas formatters and fonverters
```

**Review by datapythonista — APPROVED** [review]
Really nice, couple of suggestions, but looks great.

File: pandas/core/accessor.py (hunk 3 above)
Comment (dilex42) [reply pandas/core/accessor.py]: It is in one line.

File: pandas/plotting/_misc.py (hunk 6 above)
Comment (dilex42) [reply pandas/plotting/_misc.py]: Is it not the case of start of a sentence?

File: pandas/core/arrays/categorical.py [hunk 8]
```diff
@@ -901,11 +901,11 @@ def rename_categories(self, new_categories, inplace=False):
...
-        reorder_categories
-        add_categories
-        remove_categories
-        remove_unused_categories
-        set_categories
+        reorder_categories : Reorder categories.
+        add_categories : Add new categories.
+        remove_categories : Remove the specified categories.
+        remove_unused_categories : Remove categories which are not used.
+        set_categories : Set the categories to the specified categories.
```
Comment (dilex42) [inline pandas/core/arrays/categorical.py]:
```suggestion
        set_categories : Set the categories to the specified ones.
```

File: pandas/core/arrays/categorical.py [hunk 9]
```diff
@@ -969,11 +969,11 @@ def reorder_categories(self, new_categories, ordered=None, inplace=False):
...
-        rename_categories
-        add_categories
-        remove_categories
-        remove_unused_categories
-        set_categories
+        rename_categories : Rename categories.
+        add_categories : Add new categories.
+        remove_categories : Remove the specified categories.
+        remove_unused_categories : Remove categories which are not used.
+        set_categories : Set the categories to the specified categories.
```
Comment (dilex42) [inline pandas/core/arrays/categorical.py]:
```suggestion
        set_categories : Set the categories to the specified ones.
```

File: pandas/core/arrays/categorical.py [hunk 10]
```diff
@@ -1009,11 +1009,11 @@ def add_categories(self, new_categories, inplace=False):
...
-        rename_categories
-        reorder_categories
-        remove_categories
-        remove_unused_categories
-        set_categories
+        rename_categories : Rename categories.
+        reorder_categories : Reorder categories.
+        remove_categories : Remove the specified categories.
+        remove_unused_categories : Remove categories which are not used.
+        set_categories : Set the categories to the specified categories.
```
Comment (dilex42) [inline pandas/core/arrays/categorical.py]:
```suggestion
        set_categories : Set the categories to the specified ones.
```

File: pandas/core/arrays/categorical.py [hunk 11]
```diff
@@ -1058,11 +1058,11 @@ def remove_categories(self, removals, inplace=False):
...
-        rename_categories
-        reorder_categories
-        add_categories
-        remove_unused_categories
-        set_categories
+        rename_categories : Rename categories.
+        reorder_categories : Reorder categories.
+        add_categories : Add new categories.
+        remove_unused_categories : Remove categories which are not used.
+        set_categories : Set the categories to the specified categories.
```
Comment (dilex42) [inline pandas/core/arrays/categorical.py]:
```suggestion
        set_categories : Set the categories to the specified ones.
```

File: pandas/core/arrays/categorical.py [hunk 12]
```diff
@@ -1100,11 +1100,11 @@ def remove_unused_categories(self, inplace=False):
...
-        rename_categories
-        reorder_categories
-        add_categories
-        remove_categories
-        set_categories
+        rename_categories : Rename categories.
+        reorder_categories : Reorder categories.
+        add_categories : Add new categories.
+        remove_categories : Remove the specified categories.
+        set_categories : Set the categories to the specified categories.
```
Comment (dilex42) [inline pandas/core/arrays/categorical.py]:
```suggestion
        set_categories : Set the categories to the specified ones.
```

File: pandas/plotting/_misc.py (hunk 6 above)
Comment (datapythonista) [reply pandas/plotting/_misc.py]: Our standard, not enforced much so far, is to use `pandas` always lowercase, like a trademark. Not sure where it comes from, but we already have a PR that will validate it's always lowercase in the web and the docs, and we'll follow up with one for the code, so if you can fix it here that would be great (otherwise we'll fix it later).

File: pandas/plotting/_misc.py [hunk 13]
```diff
@@ -62,7 +62,7 @@ def deregister():
 
     See Also
     --------
-    register_matplotlib_converters : Register pandas Formatters and Converters
+    register_matplotlib_converters : Register pandas formatters and fonverters
```
Comment (datapythonista) [inline pandas/plotting/_misc.py]: typo
```suggestion
    register_matplotlib_converters : Register pandas formatters and converters
```

File: pandas/core/accessor.py (hunk 3 above)
Comment (datapythonista) [reply pandas/core/accessor.py]: I mean this seems better:
```
@doc(_register_accessor, klass="DataFrame")
def register_dataframe_accessor(name):
   
```

**Comment by dilex42** [comment]
Should be good now.

**Review by datapythonista — APPROVED** [review]
Looks perfect, thanks @dilex42

**Comment by WillAyd** [comment]
Thanks @dilex42
